pkg/job_controller/pod.go

```diff
@@ -396,9 +397,20 @@ func (jc *JobController) createNewPod(ctx context.Context, job interface{}, rt,
 		if err != nil {
 			return err
 		}
-		if err = jc.GangScheduler.BindPodToGang(podTemplate, entity); err != nil {
+
+		klog.V(5).Infof("gang scheduling enabled, gang scheduler name: %s, bind pod to gang: %s",
+			jc.GangScheduler.PluginName(), metaObject.GetName())
+
+		if err = jc.GangScheduler.BindPodToGang(metaObject, podTemplate, entity, rt, index); err != nil {
 			return err
 		}
+
+		// 1) assign gang scheduler name if it's empty.
+		// 2) override scheduler name if it differs from the selected gang implementation.
+		if podTemplate.Spec.SchedulerName == "" ||
+			(podTemplate.Spec.SchedulerName != "" && podTemplate.Spec.SchedulerName != jc.GangScheduler.SchedulerName()) {
+			podTemplate.Spec.SchedulerName = jc.GangScheduler.SchedulerName()
```

Comment:
is this the same as just do `podTemplate.Spec.SchedulerName = jc.GangScheduler.SchedulerName()
Write Preview` without the if condition ?

Reply:
done, override pod.spec.schedulerName directly.